routing/router.go

```diff
@@ -1748,16 +1748,86 @@ type routingMsg struct {
 	err chan error
 }
 
+// RouteRequest contains the parameters for a pathfinding request.
+type RouteRequest struct {
+	// Source is the node that the path originates from.
+	Source route.Vertex
+
+	// Target is the node that the path terminates at.
+	Target *route.Vertex
+
+	// Amount is the amount in millisatoshis to be delivered to the target
+	// node.
+	Amount lnwire.MilliSatoshi
+
+	// TimePreference expresses the caller's time preference for
+	// pathfinding.
+	TimePreference float64
+
+	// Restrictions provides a set of additional restrictions that the
+	// route must adhere to.
+	Restrictions *RestrictParams
+
+	// CustomRecords is a set of custom tlv records to include for the
+	// final hop.
+	CustomRecords record.CustomSet
+
+	// RouteHints contains an additional set of edges to include in our
+	// view of the graph.
+	RouteHints map[route.Vertex][]*channeldb.CachedEdgePolicy
+
+	// FinalExpiry is the cltv delta for the final hop.
+	FinalExpiry uint16
+}
+
+// NewRouteRequest produces a new route request.
+func NewRouteRequest(source route.Vertex, target *route.Vertex,
+	amount lnwire.MilliSatoshi, timePref float64,
+	restrictions *RestrictParams, customRecords record.CustomSet,
+	routeHints map[route.Vertex][]*channeldb.CachedEdgePolicy,
+	finalExpiry uint16) *RouteRequest {
+
+	return &RouteRequest{
+		Source:         source,
+		Target:         target,
+		Amount:         amount,
+		TimePreference: timePref,
+		Restrictions:   restrictions,
+		CustomRecords:  customRecords,
+		RouteHints:     routeHints,
+		FinalExpiry:    finalExpiry,
+	}
+}
+
+// validate ensures that a route request is logically sane.
+func (r *RouteRequest) validate() error {
+	if r.Target == nil {
+		return errors.New("target node required")
+	}
+
+	return nil
+}
+
+func (r *RouteRequest) target() route.Vertex {
```

Comment:
Oversight, I should unexport the fields as well. 

The accessors here are meant to "protect" against incorrect usage of the fields. Going to look at populating the fields on creation (since we can guard with a constructor to ensure the target is set from the blinded route etc) rather than accessors, because they're not _really_ protecting against direct use.